deepchem/models/torch_models/modular.py

```diff
@@ -0,0 +1,300 @@
+    ... 'activation': torch.nn.ReLU(), 'head': torch.nn.Linear(n_hidden, pt_tasks)}
+    >>> pretrain_model = torch.nn.Sequential(pretrain_components['linear'], 
+    ... pretrain_components['activation'], pretrain_components['head'])
+    >>> pretrain_modular_model = dc.models.torch_models.modular.ModularTorchModel(pretrain_model, 
+    ... pretrain_components)
+    >>> def example_pt_loss_func(inputs, labels, weights):
+    ...     return (torch.nn.functional.mse_loss(pretrain_model(inputs), labels[0]) * weights[0]).mean()
+    >>> pretrain_modular_model.loss_func = example_pt_loss_func
+    >>> pt_loss = pretrain_modular_model.fit(dataset_pt, nb_epoch=1)
+    >>> modular_model.load_from_modular(pretrain_modular_model, components=['linear'])
+    >>> ft_loss = modular_model.fit(dataset_ft, nb_epoch=1)
+
+    """
+
+    def __init__(self, model: nn.Module, components: dict, **kwargs):
+        """Create a ModularTorchModel.
+
+        Parameters
+        ----------
+        model: nn.Module
+            The model to be trained.
+        components: dict
+            A dictionary of the components of the model. The keys are the names of the
+            components and the values are the components themselves.
+        """
+
+        self.model = model
+        self.components = components
+        super().__init__(self.model, self.loss_func, **kwargs)
+        self.model.to(self.device)
+        self.components = {
+            k: v.to(self.device) for k, v in self.components.items()
+        }
+
+    def build_model(self) -> nn.Module:
+        """Builds the final model from the components."""
+        raise NotImplementedError("Subclass must define the components")
+
+    def build_components(self) -> dict:
+        """Creates the components dictionary, with the keys being the names of the
+        components and the values being torch.nn.module objects."""
+        raise NotImplementedError("Subclass must define the components")
+
+    def loss_func(self, inputs: OneOrMany[torch.Tensor], labels: Sequence,
+                  weights: Sequence) -> torch.Tensor:
+        """Defines the loss function for the model which can access the components
+        using self.components. The loss function should take the inputs, labels, and
+        weights as arguments and return the loss."""
+        raise NotImplementedError("Subclass must define the loss function")
+
+    def freeze_components(self,
+                          components: list,
+                          unfreeze: Optional[bool] = False):
+        """Freezes or unfreezes the parameters of the specified components."""
+        for component in components:
+            for param in self.components[component].parameters():
+                if unfreeze:
+                    param.requires_grad = True
+                else:
+                    param.requires_grad = False
+        self.model = self.build_model()
+        self.model.to(self.device)
+
+    def load_from_modular(self,
```

Comment:
I think it may be better to just call this `load_from_pretrained` and disable the `mypy` complaint? Feels like introducing a new API is more confusing than a slight mismatch in the call structure

Reply:
But we're still introducing a new API. We're just naming it the same as an old one.